app/webpacker/components/Panel/views/CronJobStatus/CronjobActions.jsx

```diff
@@ -0,0 +1,75 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import React, { useState } from 'react';
+import { Button } from 'semantic-ui-react';
+import runCronjob from './api/runCronjob';
+import Loading from '../../../Requests/Loading';
+import Errored from '../../../Requests/Errored';
+import resetCronjob from './api/resetCronjob';
+import { useConfirm } from '../../../../lib/providers/ConfirmProvider';
+
+export default function CronjobActions({ cronjobClassName, cronjobDetails }) {
+  const isResetDisabled = !(cronjobDetails?.in_progress && cronjobDetails?.recently_errored);
+  const isDoItDisabled = (
+    cronjobDetails?.reason_not_to_run || cronjobDetails?.scheduled || cronjobDetails?.in_progress
+  );
+
+  const [showDebugInfo, setShowDebugInfo] = useState();
+  const confirm = useConfirm();
+
+  const queryClient = useQueryClient();
+  const {
+    mutate: runCronjobMutation,
+    isLoading: isRunLoading,
+    isError: isRunError,
+  } = useMutation({
+    mutationFn: runCronjob,
+    onSuccess: (updatedCronjobDetails) => queryClient.setQueryData(
+      ['cronjob-details', cronjobClassName],
+      () => updatedCronjobDetails,
```

Comment:
In such a simple case, where you can ignore the existing data and you just want to overwrite it completely, you can pass `updatedCronjobDetails` by itself, without wrapping it into an "empty" function.